src/data/list/indexes.lean

```diff
@@ -16,6 +18,27 @@ namespace list
 
 variables {α : Type u} {β : Type v}
 
+section map_with_index
+
+lemma map_with_index_core_eq (l : list α) (f : ℕ → α → β) (n : ℕ) :
+  l.map_with_index_core f n = l.map_with_index (λ i a, f (i + n) a) :=
+begin
+  induction l with hd tl hl generalizing f n,
+  { simp [list.map_with_index, list.map_with_index_core] },
+  { rw [list.map_with_index],
+    simp [list.map_with_index_core, hl, add_left_comm, add_assoc, add_comm] }
+end
+
+lemma map_with_index_eq_enum_map_uncurry (l : list α) (f : ℕ → α → β) :
+  l.map_with_index f = l.enum.map (function.uncurry f) :=
```

Comment:
Oh one more thing: this lemma should be called `map_with_index_eq_map_enum`. The `uncurry` is incidental and the outermost operation on the right-hand side is `map`, not `enum`. (This is obfuscates by dot notation.)

Reply:
Done